pkg/lib/authn/otp/service.go

```diff
@@ -41,7 +50,11 @@ func (s *Service) createCode(target string, otpMode OTPMode, codeModel *Code) (*
 	switch otpMode {
 	case OTPModeMagicLink:
 		codeModel.Code = secretcode.MagicLinkOTPSecretCode.Generate()
-		err := s.CodeStore.Create(codeModel.Code, codeModel)
+		err := s.CodeStore.Create(target, codeModel)
+		if err != nil {
+			return nil, err
+		}
+		err = s.MagicLinkStore.Create(codeModel.Code, codeModel.Target, codeModel.ExpireAt)
```

Comment:
Given that `MagicLinkStore.Create` may failed due to duplicate code (although it is rare), we should call `MagicLinkStore.Create` first. And call `CodeStore.Create` if `MagicLinkStore.Create` succeed.

Reply:
Updated